modulus/distributed/shard_tensor.py

```diff
@@ -0,0 +1,564 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast, Optional, Tuple, Sequence, Dict, List, Union
+from collections.abc import Iterable
+
+import torch
+
+
+import torch.distributed.tensor._dispatch as op_dispatch
+
+from torch.distributed.tensor.placement_types import (
+    Partial, 
+    Placement,
+    Replicate,
+    Shard
+)
+from torch.distributed.tensor._dtensor_spec import (
+    TensorMeta,
+)
+from modulus.distributed.utils import compute_split_shapes, split_tensor_along_dim
+
+
+import torch.distributed as dist
+
+from modulus.distributed._shard_tensor_spec import (
+    ShardTensorSpec,
+    _infer_shard_tensor_spec_from_local_chunks
+)
+
+from modulus.distributed._shard_redistribute import (
+    ShardRedistribute
+)
+from torch.distributed.device_mesh import _mesh_resources, DeviceMesh
+
+
+from modulus.distributed import DistributedManager
+class _ToTorchTensor(torch.autograd.Function):
```

Comment:
Formatting (add newline between import and class def). Did linting not catch this?